### PR TITLE
AC: Added groups of arguments for AC

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 1]
```diff
@@ -38,145 +38,266 @@
...
-    parser.add_argument(
-        '-b', '--bitstreams',
-        help='prefix path to bitstreams folder',
-        type=partial(get_path, file_or_directory=True),
+    common_args.add_argument(
+        '-tt', '--target_tags',
+        help='space separated list of launcher tags for infer',
+        required=False,
+        nargs='+'
+    )
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:96]: I think it should be added to separated group - filtration

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 2]
```diff
@@ -38,145 +38,266 @@
...
         required=False
     )
-    parser.add_argument(
+    common_args.add_argument(
+        '--model_attributes',
+        help="path's prefix for additional models attributes",
+        type=partial(get_path, is_directory=True),
+        required=False
+    )
+    common_args.add_argument(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:108]: it is OV specific, not common

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 3]
```diff
@@ -38,145 +38,266 @@
...
         '-a', '--annotations',
         help='prefix path to the converted annotations and datasets meta data',
         type=partial(get_path, is_directory=True),
         required=False
     )
-    parser.add_argument(
-        '-e', '--extensions',
-        help='prefix path to extensions folder',
-        type=partial(get_path, check_exists=False),
+    common_args.add_argument(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:73]: it is MO specific

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 4]
```diff
@@ -37,147 +37,279 @@
...
+        )
+
+    parser = ArgumentParser(description='Deep Learning accuracy validation framework', allow_abbrev=False)
+    add_common_args(parser)
+    add_dataset_related_args(parser)
+    add_profiling_related_args(parser)
+    add_tool_customization_args(parser)
+    add_openvino_specific_args(parser)
 
     parser.add_argument(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:314]: it is OV specific (hetero plugin)

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 5]
```diff
@@ -186,7 +318,6 @@ def build_arguments_parser():
         default=Path.cwd(),
         required=False
     )
-
     parser.add_argument(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:321]: it is OV specific (FPGA)

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 6]
```diff
@@ -196,94 +327,21 @@ def build_arguments_parser():
...
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:327]: it is OV related (myriad, HDDL are VPU devices)

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 7]
```diff
@@ -196,94 +327,21 @@ def build_arguments_parser():
...
-        '--shuffle', help="Allow shuffle annotation during creation a subset",
-        required=False,
-        type=cast_to_bool
+        '--version',
+        help='show tool version and exit',
+        action='version',
+        version='%(prog)s {version}'.format(version=__version__)
     )
     parser.add_argument(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:340]: it is tool argument

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 8]
```diff
@@ -37,147 +37,279 @@
...
+            required=False
+        )
+        tool_customization_args.add_argument(
+            '--metrics_interval',
+            help='number of iteration for updated metrics result printing',
+            type=int,
+            default=1000,
+            required=False
+        )
+        tool_customization_args.add_argument(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:220]: only ov support async mode now, so make sense to move OV section async related args

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 9]
```diff
@@ -37,147 +37,279 @@
...
 
-    parser.add_argument(
-        '--ignore_result_formatting',
-        help='allow to get raw metrics results without data formatting',
-        required=False,
-        default=False,
-        type=cast_to_bool
-    )
+    def add_tool_customization_args(parser):
+        tool_customization_args = parser.add_argument_group('Tool customization arguments')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:174]: for me castomization means that you modify running tool, but we just setup some logging (results, progress and so on)
better
```suggestion
        tool_customization_args = parser.add_argument_group('Tool settings arguments')
```
or logging arguments
BTW, log file should be also in this category

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 1 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py]: Added

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 2 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py:83]: Moved

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 3 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py:72]: Moved

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 4 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py]: Moved

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 5 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py]: Moved

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 6 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py]: Moved

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 7 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py]: Moved

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 8 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py]: Moved

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 9 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py]: Renamed

**Comment by Anna-Mironova** [comment]
Jenkins please retry a build

**Review by eaidova — APPROVED** [review]

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 10]
```diff
@@ -36,254 +36,322 @@
...
         help='prefix path to the converted annotations and datasets meta data',
         type=partial(get_path, is_directory=True),
         required=False
     )
-    parser.add_argument(
-        '-e', '--extensions',
-        help='prefix path to extensions folder',
-        type=partial(get_path, check_exists=False),
-        default=Path.cwd(),
+    common_args.add_argument(
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:72]: please move logging file to tool settings arguments

File: tools/accuracy_checker/accuracy_checker/main.py [hunk 11]
```diff
@@ -36,254 +36,322 @@
...
+    config_filtration_args.add_argument(
+        '-tt', '--target_tags',
+        help='space separated list of launcher tags for infer',
+        required=False,
+        nargs='+'
+    )
+
+
+def add_dataset_related_args(parser):
+    dataset_related_args = parser.add_argument_group('Datasets-related arguments')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/main.py:115]:
```suggestion
    dataset_related_args = parser.add_argument_group('Dataset related arguments')
```

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 10 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py:72]: Moved

File: tools/accuracy_checker/accuracy_checker/main.py (hunk 11 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/main.py]: Done

**Pull request merged**